bundles/org.openhab.core.addon.eclipse/src/main/java/org/openhab/core/addon/eclipse/internal/EclipseAddonService.java

```diff
@@ -0,0 +1,188 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.addon.eclipse.internal;
+
+import static java.util.Map.entry;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.addon.Addon;
+import org.openhab.core.addon.AddonInfo;
+import org.openhab.core.addon.AddonInfoRegistry;
+import org.openhab.core.addon.AddonService;
+import org.openhab.core.addon.AddonType;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * This is an implementation of an {@link AddonService} that can be used when debugging in Eclipse.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+@Component
+@NonNullByDefault
+public class EclipseAddonService implements AddonService {
+    private static final String ADDONS_CONTENT_TYPE = "application/vnd.openhab.bundle";
+    private static final String ADDONS_AUTHOR = "openHAB";
+    private static final String BUNDLE_SYMBOLIC_NAME_PREFIX = "org.openhab.";
+
+    private static final String ADDON_TYPE_AUTOMATION = "automation";
+    private static final String ADDON_TYPE_BINDING = "binding";
+    private static final String ADDON_TYPE_MISC = "misc";
+    private static final String ADDON_TYPE_PERSISTENCE = "persistence";
+    private static final String ADDON_TYPE_TRANSFORMATION = "transformation";
+    private static final String ADDON_TYPE_UI = "ui";
+    private static final String ADDON_TYPE_VOICE = "voice";
+
+    private static final List<AddonType> ADDON_TYPES = List.of( //
+            new AddonType(ADDON_TYPE_AUTOMATION, "Automation"), //
+            new AddonType(ADDON_TYPE_BINDING, "Bindings"), //
+            new AddonType(ADDON_TYPE_MISC, "Misc"), //
+            new AddonType(ADDON_TYPE_VOICE, "Voice"), //
+            new AddonType(ADDON_TYPE_PERSISTENCE, "Persistence"), //
+            new AddonType(ADDON_TYPE_TRANSFORMATION, "Transformations"), //
+            new AddonType(ADDON_TYPE_UI, "User Interfaces"));
+
+    private static final Map<String, String> ADDON_BUNDLE_TYPE_MAP = Map.ofEntries(
+            entry(ADDON_TYPE_AUTOMATION, "automation"), //
+            entry(ADDON_TYPE_BINDING, "binding"), //
+            entry(ADDON_TYPE_MISC, "io"), //
+            entry(ADDON_TYPE_PERSISTENCE, "persistence"), //
+            entry(ADDON_TYPE_TRANSFORMATION, "transform"), //
+            entry(ADDON_TYPE_VOICE, "voice"));
+
```

Comment:
This is duplicated in almost all add-on services. Wouldn't it make sense to provide these as "default" from `AddonType`?

Reply:
Yes I don't like all the duplication either so that might be something to cleanup as well.
I first wanted to quickly check if this approach would turn into something usable. :wink: